packages/bc-wallet-adapter-client-api/src/util/CypherUtil.ts

```
@@ -0,0 +1,235 @@
+import bs58 from 'bs58'
+import { Buffer } from 'buffer'
+import crypto from 'crypto'
+
+import { environment } from '../environment'
+
+// TODO move to common package to dedup
+
+const env = environment.encryption
+
+function isChaCha20Poly1305Supported(): boolean {
+  try {
+    const supportedAlgorithms = crypto.getCiphers()
```

only catch this line? else we are eating all errors?

---

fixed